### PR TITLE
Amazon Seller Partner: Validate connections without orders data

### PR DESCRIPTION
**Pull request by animer3009**
## What
Not all accounts have orders and not all reports are orders based.
So checking connections by checking existing orders is not a good idea.

Described issue: https://github.com/airbytehq/airbyte/issues/20669

## How
Instead of raising error we are validating connections where checking Orders stream ended with `StopIteration` error.

### CODE REVIEW

**Comment by animer3009** [comment]
@bazarnov @monai Can you guys review and approve current changes?

**Review by bazarnov — APPROVED** [review]
Lgtm, but integration tests should be passed as well as acceptance tests.

@marcosmarxm tagging you here to follow up, after the code freeze ends.

**Comment by animer3009** [comment]
Guys, I need to do something or it will auto merged?

**Comment by animer3009** [comment]
@marcosmarxm @bazarnov Why is this merge is stack so long?

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py [hunk 1]
```diff
@@ -140,9 +140,9 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
             return True, None
         except Exception as e:
             if isinstance(e, StopIteration):
-                logger.error(
-                    "Could not check connection without data for Orders stream. Please change value for replication start date field."
-                )
+                # Validate Orders stream without data
+                if isinstance(e, StopIteration):
```
[the remark this answers is not in the document]
Comment (animer3009) [reply airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py]: @sajarin Sure! you are right. I am not sure why it happened, but I pushed fix :)

**Review by sajarin — APPROVED** [review]

**Pull request merged**